### ISSUE TITLE
Attention map visualization on the CIFAR100

### ISSUE THREAD

**Issue by mymuli** [machine-perception-robotics-group/attention_branch_network#7]
Hi， Thank you for sharing the good idea and code.

On CIFAR 100, how can I use the training results to visualize the sample images?

**Comment by Hiroshi-Fukui** [comment]
Thanks for the repo.
If you want to visualize an input image (and attention map), please refer to the 334~344 from imagenet.py.
Note that, imagenet.py visualizes the input image in evaluating only. I think that if you apply the visualizing process at imagenet.py to cifar100 at the training, you can visualize the input image at the training.

**Comment by mymuli** [comment]
@Hiroshi-Fukui 
Dear Sir:
     First of all, thank you very much for your reply！

Following your suggestion, I used the relevant code snippet of file imagene.py to get the following visualization results during the **evaluating phase**.


![demo-1](https://user-images.githubusercontent.com/37521721/62750923-5a095d00-ba94-11e9-8e82-bdd3d7ec735c.png)
![demo1-2](https://user-images.githubusercontent.com/37521721/62750927-6097d480-ba94-11e9-82f8-0aff9bb50794.png)
![demo2](https://user-images.githubusercontent.com/37521721/62750934-655c8880-ba94-11e9-8aab-07ed70e5434a.png)
![demo2-2](https://user-images.githubusercontent.com/37521721/62750937-67264c00-ba94-11e9-9902-ae33e3bed512.png)

Although effective, the results should not be the best. Since the image size of the cifar100 data set is **different** from that of the imagene data set, can you give me some suggestions on how to modify the following lines of code?
`
 v_img = ((item_img.transpose((1,2,0)) + 0.5 + [0.485, 0.456, 0.406]) * [0.229, 0.224, 0.225])* **256**
 
v_img = v_img[:, :, ::-1]

 resize_att = cv2.resize(item_att[0], (in_x, in_y))

 resize_att *= **255**.
`

**Comment by Hiroshi-Fukui** [comment]
You don't need to adjust the image size for cifar and imagenet due to adjusting the image size with automatic. (Please get the image size using ```in_b, in_c, in_y, in_x = inputs.shape```)

Additionally, I think that you need to adjust the number of average and variance. Average and variance on cifar and imagenet differ.

v_img = ((item_img.transpose((1,2,0)) + 0.5 + [0.485, 0.456, 0.406]) * [0.229, 0.224, 0.225])* 256
↑
Imagenet: ave.=[0.485, 0.456, 0.406], var.=[0.229, 0.224, 0.225]
cifar: ave.=[0.4914, 0.4822, 0.4465], var.=[0.2023, 0.1994, 0.2010]

**Comment by mymuli** [comment]
@Hiroshi-Fukui 
Thank you again！
If I want to try on my own dataset, how should I structure my dataset for this? What aspects do I need to adjust?

**Comment by Hiroshi-Fukui** [comment]
If you use your dataset, I think that better modifying the code based on imagenet.py.
imagenet.py uses ```torch.utils.data.DataLoader``` and ```torchvision.datasets.ImageFolder```. ```torchvision.datasets.ImageFolder``` can load the images splited by directories for each category on image classification. So, if you can set the directory structure on ```torchvision.datasets.ImageFolder```, you can train the architecture by changing the dataset directory path. 

All that is left is to adjust the number of categories, average and variance, image size, training parameters (i.e. epochs, learning rate, ...).

**Comment by mymuli** [comment]
@Hiroshi-Fukui 
Thank you very much. 
Besides, if I want to use it in **multi-label classification**, what parts of the program need to be modified?
